src/anemoi/inference/grib/templates/__init__.py

```diff
@@ -75,21 +80,24 @@ def template(self, variable: str, lookup: dict[str, Any]) -> ekd.Field:
 class IndexTemplateProvider(TemplateProvider):
     """Template provider based on an index file."""
 
-    def __init__(self, manager: Any, index_path: str) -> None:
+    def __init__(self, manager: Any, index_path: str | list) -> None:
```

Comment:
Would it be possible to replace `Any` with something else? A base class? Protocol?